exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/__init__.py

```diff
@@ -78,14 +79,13 @@
 def _convert_buckets(metric: Metric) -> Sequence[Tuple[str, int]]:
     buckets = []
     total_count = 0
-    for index, value in enumerate(metric.point.bucket_counts):
-        total_count += value
-        buckets.append(
-            (
-                f"{metric.point.explicit_bounds[index]}",
-                total_count,
-            )
-        )
+    for upper_bound, count in zip(
+        chain(metric.point.explicit_bounds, ["+Inf"]),
+        metric.point.bucket_counts,
+    ):
+        total_count += count
```

Comment:
According to the [data model](https://github.com/open-telemetry/opentelemetry-proto/blob/0e254b5c04614244edb35a9d4bf73ca26b44438a/opentelemetry/proto/metrics/v1/metrics.proto#L577) should histograms already have a total count property? Maybe we can let the sdk keep track of this value and just use that?

Reply:
Right, but this is used to get intermediate counts as well which are what prometheus expects. Each prometheus histogram bucket includes all the counts *up to* that bucket (`(-inf, b1], (-inf, b2], (-inf, b3], ...`).

Reply:
EDIT: I misunderstood the histogram behaviour of prometheus, so ignore the previous comment. Is it specific to prometheus that the counts for a certain bucket index i represent ALL values that are less than the value at i, not just the ones greater than value at i-1 and less than i? (less than or equal)

Reply:
EDIT 2: Nevermind found this: https://www.robustperception.io/why-are-prometheus-histograms-cumulative